src/JukeBox.cpp

```diff
@@ -33,46 +50,205 @@ bool JukeBox::start(string exeName, string configFilePath) {
   return true;
 }
 
-TResult<TSessionID> JukeBox::generateSession(
-    optional<TPassword> const &pw, std::optional<std::string> const &nickname) {
-  return Error(ErrorCode::NotImplemented,
-               "generateSession is not implemented yet");
+TResult<TSessionID> JukeBox::generateSession(optional<TPassword> const &pw,
+                                             optional<string> const &nickname) {
+  static int userID = 0;
+  User user;
+  auto conf = ConfigHandler::getInstance();
+  auto adminPw = conf->getValueString("MainParams", "adminPassword");
+  if (holds_alternative<Error>(adminPw))
+    return get<Error>(adminPw);
+
+  string name = "(no nickname given)";
+  if (nickname.has_value()) {
+    name = nickname.value();
+    user.Name = name;
+  }
```

Comment:
if `nickname` is empty, `user.Name` will be an empty string (the default value). is this intended?